src/expressions/lessThanExpressionChecker.js

```diff
@@ -19,11 +19,11 @@ export class LessThanExpressionChecker extends BaseSubExpressionChecker {
 
       lastOperandResult = operandResult
 
-      if (typeof operandResult === 'undefined') {
-        hasUndefined = true
+      if (!BigUtil.isNumeric(operandResult)) {
+        hasNonNumeric = true
       }
     }
-    const lessThanResult = hasUndefined ? undefined : true
+    const lessThanResult = hasNonNumeric ? undefined : true
```

Comment:
So, while I'm not opposed to this per se, we do need to consider the consequences here.

Before this "undefined" means "I don't know for sure", and should only occur when evaluating potential actions. 

Here, we overload it with the meaning that it's not numeric, so trying to take an action could result in resolution to undefined.

How about always resolving non-numeric comparisons to false?